test/Tests/src/Random_Spec.enso

```diff
@@ -1,79 +1,154 @@
+        all_from_small_range = [Date.new 2023 03 01, Date.new 2023 03 02, Date.new 2023 03 03]
+        dates = 0.up_to 100 . map (_-> Random.date small_range)
+        dates.should_contain_the_same_elements_as all_from_small_range
+
+    Test.specify "should allow generating random times" <|
+        Random.set_seed 12345
+        start = Time_Of_Day.new 8 12 15
+        end = Time_Of_Day.new 9 40 2
+        Random.time start end . should_equal (Time_Of_Day.new 9 31 10)
+        Random.time start end . should_equal (Time_Of_Day.new 8 34 31)
+        Random.time start end . should_equal (Time_Of_Day.new 9 27 48)
+        Random.time . should_equal (Time_Of_Day.new 0 13 48)
+        Random.time . should_equal (Time_Of_Day.new 14 20 55)
+        Random.time . should_equal (Time_Of_Day.new 8 48 4)
+
+        small_range_start = Time_Of_Day.new 8 12 15
+        small_range_end = Time_Of_Day.new 8 12 18
+        all_from_small_range = [Time_Of_Day.new 8 12 15, Time_Of_Day.new 8 12 16, Time_Of_Day.new 8 12 17, Time_Of_Day.new 8 12 18]
+        times = 0.up_to 100 . map (_-> Random.time small_range_start small_range_end)
+        times.should_contain_the_same_elements_as all_from_small_range
+
+    Test.specify "should allow generating random UUIDs" <|
+        Random.uuid . should_only_contain_elements_in "0123456789abcdef-"
+
+    Test.specify "should allow to select a random item from a vector, without replacement" <|
+        Random.set_seed 0
         vector = ["A", "B", "C"]
         shuffles = 0.up_to 100 . map _->
-            Random.sample vector 2 rng
+            Random.item vector 2
         shuffles . should_contain_the_same_elements_as [["A", "B"], ["A", "C"], ["B", "A"], ["B", "C"], ["C", "A"], ["C", "B"]]
 
-        overflow = Random.sample vector 100 rng
+        overflow = Random.item vector 100
         overflow.length . should_equal 3
         overflow.should_contain_the_same_elements_as vector
 
-        Random.sample ["A", "A", "A"] 2 rng . should_equal ["A", "A"]
-        Random.sample ["A", "A", "A"] 0 rng . should_equal []
-        Random.sample ["A", "A", "A"] 3 rng . should_equal ["A", "A", "A"]
-        Random.sample ["A", "A", "A"] 100 rng . should_equal ["A", "A", "A"]
+        Random.item ["A", "A", "A"] 2 . should_equal ["A", "A"]
+        Random.item ["A", "A", "A"] 0 . should_equal []
+        Random.item ["A", "A", "A"] 3 . should_equal ["A", "A", "A"]
+        Random.item ["A", "A", "A"] 100 . should_equal ["A", "A", "A"]
+
+    Test.specify "should allow to select a random item from a vector, with replacement" <|
+        Random.set_seed 0
+        vector = 0.up_to 10 . to_vector
+
+        Random.item vector . length . should_equal 1
+        Random.item vector 4 . length . should_equal 4
+
+        0.up_to 20 . map _->
+            items = Random.item vector 5 with_replacement=True
+            items.should_only_contain_elements_in vector
```

Comment:
I think we should also check the length of `items`.

The current test can be 'trivially' satisfied by `Random.items` returning `[]` but that is not what we want.

Reply:
And btw. I appreciate adding the new `should_` matcher to the test suite - it makes for easier to understand error messages if some cases fail.

Reply:
Done